### ISSUE TITLE
PDF form text field setText does not use font

### ISSUE THREAD

**Issue by psyanite** [Hopding/pdf-lib#1378]
Hi there, thanks for creating the best PDF library on the internet! 
I ran into some issues trying to fill text fields on my AcroForm

### What were you trying to do?

I tried to use PDF-LIB `form.getTextField('dateTaken').setText('hello this is my text')` to set some text fields on my form.
But the fonts I configured in Adobe Acrobat Pro is ignored.

I created a form using Adobe Acrobat Pro
When creating the fields in Adobe Acrobat Pro I can edit the text field properties and set a font.
<img width="569" alt="Screen Shot 2023-01-13 at 10 29 44 am" src="https://user-images.githubusercontent.com/24215698/212203004-bda8c0fe-57e2-43f3-881a-97fd2665a4b4.png">

When I fill the form in Adobe Acrobat it fills correctly, the font size, color, and font is maintained:
<img width="767" alt="Screen Shot 2023-01-13 at 10 25 46 am" src="https://user-images.githubusercontent.com/24215698/212203213-2ff0c56d-2bb4-4884-93db-859fc2b5999d.png">

But when I use the following code: 
```
      form.getTextField('dateTaken').setText('hello this is my text')
      const pdfBytes = await pdfDoc.save()
      download(pdfBytes, "result.pdf", "application/pdf")
```
And save the pdf, the font is not maintained:
<img width="736" alt="Screen Shot 2023-01-13 at 10 26 49 am" src="https://user-images.githubusercontent.com/24215698/212203372-c2bce6c5-0d53-4bfe-9d7b-e0e495572eff.png">

### How did you attempt to do it?

```
  <head>
    <meta charset="utf-8" />
    <script src="https://unpkg.com/pdf-lib@1.11.0"></script>
    <script src="https://unpkg.com/downloadjs@1.4.7"></script>
  </head>

<script>
    const { PDFDocument } = PDFLib
    async function fillForm() {

      var myPdfInBytes = await document.getElementById("mypdf").files[0].arrayBuffer()
      const pdfDoc = await PDFDocument.load(myPdfInBytes)
      const form = pdfDoc.getForm()

      // Fill text fields
      form.getTextField('dateTaken').setText('hello this is my text')

      const pdfBytes = await pdfDoc.save()
      download(pdfBytes, "result.pdf", "application/pdf")
    }
  </script>
```

### What actually happened?

The generated PDF shows the correct text, the correct font color and font size, but not the correct font. 

All fonts are embedded:
<img width="306" alt="Screen Shot 2023-01-13 at 10 36 02 am" src="https://user-images.githubusercontent.com/24215698/212203725-373587fc-9fd5-400c-8a9a-16af3bf066da.png">

### What did you expect to happen?

I expected the text fields to be in the correct font, like when I filled the form out using Adobe Acrobat.

### How can we reproduce the issue?

JSFIDDLE:
https://jsfiddle.net/psyanite/9vdeq2nt/35/

Test PDF:
[test.pdf](https://github.com/Hopding/pdf-lib/files/10406919/test.pdf)

### Version

latest version

### What environment are you running pdf-lib in?

Browser

### Checklist

- [X] My report includes a Short, Self Contained, Correct (Compilable) Example.
- [X] I have attached all PDFs, images, and other files needed to run my SSCCE.

**Comment by pietrondo** [comment]
same problems... there is a solution?

**Comment by coffedahl** [comment]
Edit: 
I found the problem when i scanned trough the code i think. 

When the ```PDFTextField.setText('string')``` command i called it markes the field as "dirty" that somehow tells the library that the field needs updating in styling. 
If you tried to add the font by yourself with ```PDFDocument.embedFont(font)``` and then try to update the apperance with ```PDFTextField.updateAppearances()``` it still marks it as dirty and changes the font.

The solution i found was that the ```updateAppearances(font)``` command marks it as clean again so as long as you call the ```updateAppearances()``` after you call the ```setText()```  the field is marked as clean and the program wont change your font

I dont think i have the coding skills neccessary to make a fix for it but this is a half solution until someone smarter can fix it :)

## Example
```javascript
// Imports
const { PDFDocument } = require('pdf-lib')
const { readFile } = require('fs/promises')
const fontkit = require('@pdf-lib/fontkit')
// Import pdf document
const document = await PDFDocument.load(await readFile('/path/to/file.pdf'))
// Create a custom font
document.registerFontkit(fontkit)
const custonFont = await document.embedFont(await readFile('/custom/font.ttf'))
// Get form and field
const form = document.getForm()
const titleField = form.getTextField('form identifier')
// Set the text and after update to font
titleField.setText(object.title)
titleField.updateAppearances(custonFont)
```

**Comment by yepMad** [comment]
I was doing `textField.updateAppearances(font)` and then `textField.setAlignment(alignment)`. This caused the same problem. So I believe the final word is: just use `updateAppearances` after you did all setters

**Comment by gioppoluca** [comment]
I do not see the needed font even after the `updateAppearances` in the pdf the field is marked as using the right font, but opening the file I do not see the right font.

**Comment by tacomanator** [comment]
I'm having similar problem. Another issue mentioned it working only after `form.flatten()` is called. Unfortunately this makes the fields uneditable (additionally I am facing issue where text is garbled when using `{subset: true}` when embedding the font, but it's probably not related to use of flatten).

**Comment by morbalint** [comment]
I'm having the same issue. Thanks coffedahl for the workaround. 
FYI: calling `form.updateFieldAppearances(pdfFont)` on the entire form also solved the issue and maybe easier to achieve in some cases.

**Comment by p3pp8** [comment]
Hello, 

Facing same issue, i don't understand why if i have a font already embedded and used for the whole PDF and its forms fields i have to embed the same font again, this is clearly a BUG. 
Anyone knows how to force form text fields to use the same font and style already set for the whole PDF and its form fields without embed another one? The thing about dirty and clear is riducolous, why the lib should force using Helvetica when form fields already have their style set?

**Comment by Roxcsas** [comment]
Same issue, when i set text the value of the font get overwritten with Helvetica as stated in the docs, there is any way to keep the pdf font value?

**Comment by josephine-njooi** [comment]
+1
I can't get the font to change either through `field.updateAppearances(pdfFont)`,  `field.defaultUpdateAppearances(pdfFont)`, or `form.updateFieldAppearances(pdfFont)`, whether I do it before or after setText.
